djangocms_link/forms.py

```diff
@@ -18,7 +18,7 @@ def for_site(self, site):
         # current site
         # this will work for PageSelectFormField
         from cms.models import Page
-        self.fields['internal_link'].queryset = Page.objects.drafts().on_site(site)
+        self.fields['internal_link'].queryset = Page.objects.on_site(site)
```

Comment:
Can we try copying this compatibility for djangocms 4 into a compatibility file in this project similar to this: djangocms_url_manager/compat.py

When we have that compatibility check we can do and if else here to keep the package django cms 3 and 4 compatible.

Reply:
Okay. But idk what too do now 😅

Reply:
Create a new file 'compat.py' with the contents from the link above 'CMS_LT_4'.

Import 'CMS_LT_4' in this forms.py file and have:

If CMS_LT_4:
    Use the code that you deleted
Else
    Use the code you added.

The above makes the change all cms versions compatible.

Reply:
Okay. And how can i see, that it works?. And do i have to commit those changes with the compat.py file?.

Reply:
Is that okay?:

```python
from django.forms.models import ModelForm
from django.utils.translation import gettext_lazy as _

from djangocms_attributes_field.widgets import AttributesWidget

from .fields import PageSearchField
from .models import Link

import 'CMS_LT_4'

if CMS_LT_4:
    self.fields['internal_link'].queryset = Page.objects.drafts().on_site(site)
    else
    self.fields['internal_link'].queryset = Page.objects.on_site(site)
```

Reply:
That looks ok, the import would be `from compat import CMS_LT_4`

Maybe add a comment above CMS_LT_4 to state this is for django cms 4:

```
# django CMS 3 compatibility
if CMS_LT_4:
    ...
# django CMS 4.0.x + compatibility
else:
    ...
```

Reply:
![114300171-35625580-9abf-11eb-9794-21d7799b6ba3](https://user-images.githubusercontent.com/39345336/114300182-4743f880-9abf-11eb-81f0-a92dd713e2a5.png)

Reply:
And this is in my `compat.py`
![Bildschirmfoto 2021-04-11 um 12 14 17](https://user-images.githubusercontent.com/39345336/114300223-722e4c80-9abf-11eb-96ed-3dfb7d7514ff.png)

Reply:
Just to be sure, this is the content of my `compat.py` file

```python
from distutils.version import LooseVersion

import cms


CMS_VERSION = cms.__version__

CMS_LT_4 = LooseVersion(CMS_VERSION) < LooseVersion("4.0")


def get_page_placeholders(page, language=None):
    try:
        # cms3.6 compat
        return page.get_placeholders()
    except TypeError:
        return page.get_placeholders(language)
```

Reply:
I will try It again today 👍🏻. Should i commit the Changes with the compat file too?.

Reply:
Yes commit everything that you have to this PR.

I don't think that you need this unless you're using it somewhere:
```
def get_page_placeholders(page, language=None): 
    try: 
        # cms3.6 compat 
        return page.get_placeholders()
    except TypeError: 
        return page.get_placeholders(language)
```

Reply:
Also this package has nothing to do with the Carousel Slide plugin, what you are actually doing here is adding v4 compatibility to the link package, similar to this PR: https://github.com/nephila/djangocms-redirect/pull/41/files

The title of this PR should be: "Feature: Added initial django-cms 4.0.x compatibility"

Reply:
Okay. Sorry Andrew! I will commit everything today 😊

Reply:
Its commited 😊